docs/docs/anatomy.md

```diff
@@ -0,0 +1,254 @@
+
+```ts
+export namespace Events {
+  // prettier-ignore
+  const date = z.string().datetime().transform((x) => new Date(x))
+  export const Timestamp = z.object({ timestamp: date })
+  export const ActualsOverridden = z.object({ checkedIn: date, checkedOut: date })
+  export type ActualsOverridden = z.infer<typeof ActualsOverridden>
+
+  export type Event =
+    | { type: "CheckedIn"; data: Timestamp }
+    | { type: "CheckedOut"; data: Timestamp }
+    | { type: "ActualsOverridden"; data: ActualsOverridden }
+
+  export const codec = Codec.upcast<Event>(
+    Codec.json(),
+    Codec.Upcast.from({
+      CheckedIn: Timstamp.parse,
+      CheckedOut: Timstamp.parse,
+      ActualsOverridden: ActualsOverridden.parse,
+    }),
+  )
+}
+```
+
+## The Fold
+
+```ts
+export namespace Fold {
+  import Event = Events.Event
+  export type State = { checkedIn?: Date; checkedOut?: Date }
+  export const initial: State = {}
+  export const evolve = (state: State, event: Event) => {
+    switch (event.type) {
+      case "CheckedIn":
+        return { ...state, checkedIn: event.data.timestamp }
+      case "CheckedOut":
+        return { ...state, checkedOut: event.data.timestamp }
+      case "ActualsOverridden":
+        return event.data
+    }
+  }
+}
+```
+
+## The Decide
+
+```ts
+export namespace Decide {
+  import Event = Events.Event
+  import State = Fold.State
+  export const checkIn =
+    (timestamp: Date) =>
+    (state: State): Event[] => {
+      if (state.checkedIn) {
+        // no change
+        if (+state.checkedIn === +timestamp) return []
+        throw new Error("Already checked in with different timestamp")
+      }
+      return [{ type: "CheckedIn", data: { timestamp } }]
+    }
+  export const checkOut =
+    (timestamp: Date) =>
+    (state: State): Event[] => {
+      if (state.checkedOut) {
+        // no change
+        if (+state.checkedOut === +timestamp) return []
+        throw new Error("Already checked out with different timestamp")
+      }
+      return [{ type: "CheckedOut", data: { timestamp } }]
+    }
+
+  export const manuallyOverride = (checkedIn: Date, checkedOut: Date) => (state: State) => {
+    if (+checkedIn === +state?.checkedIn && +checkedOut === +state?.checkedIn) return []
+    return [{ type: "ActualsOverridden", data: { checkedIn, checkedOut } }]
+  }
+}
+```
+
+## The Service
+
+```ts
+export class Service {
+  constructor(
+    private readonly resolve: (
+      appointmentId: AppointmentId,
+      userId: UserId,
+    ) => Decider<Events.Event, Fold.State>,
+  ) {}
+
+  checkIn(appointmentId: AppointmentId, userId: UserId, timestamp: Date) {
+    const decider = this.resolve(appointmentId, userId)
+    return decider.transact(Decide.checkIn(timestamp), LoadOption.AssumeEmpty)
+  }
+
+  checkOut(appointmentId: AppointmentId, userId: UserId, timestamp: Date) {
+    const decider = this.resolve(appointmentId, userId)
+    return decider.transact(Decide.checkOut(timestamp), LoadOption.AssumeEmpty)
```

Comment:
dont assume empty for checkOut
a) coz it wont be
b) to underscore that most of the time you dont do or think about fancy options

Reply:
```suggestion
    return decider.transact(Decide.checkOut(timestamp))
```

Reply:
yes, think so